### ISSUE TITLE
Should response be in message as a hash ?

### ISSUE THREAD

**Issue by maximede** [logstash-plugins/logstash-input-http_poller#7]
I disagree a bit with the new implementation regarding the decoding of the response body.
We are currently putting it completely in the message, as a hash.
It means that if you want to use the content present in the json/xml, you'll need to parse it in a filter.

I think it would it make sense to provide an option to parse the content of the response and add all the variables to the event, as I was doing in the initial version of this input

Thought ?

**Comment by andrewvc** [comment]
Hi Maxime! I was mostly worried that if variables such as 'url' were part of the event itself (url is a very common key) it would be overwritten, hence the nesting. 

I'm definitely open to some alternative ideas here. I'd also considered flipping it around and stuffing the metadata in '_poller_metadata' or something. What are your thoughts?

**Comment by maximede** [comment]
Actually, I've never really used nested variables with the elk stack. How well does it work ? 
Maybe we could prefix all the variables the plugin is adding with 'http_poller' ?
I don't really now what the best practices are regarding this

**Comment by andrewvc** [comment]
In terms of efficiency and usability. In Logstash, elasticsearch and kibana nested values are handled just as efficiently and easily as first level values since they all treat data as JSON-like hashes, arrays, and scalar values. They are just as query-able and efficient.

That being said, I think I think it makes sense to get some feedback from the rest of the logstash team on this tomorrow when they're back at work. The important thing here is consistency and convention in relation to other plugins. If there isn't a best practice here there should be! This is a common enough situation that we shouldn't have to make up the rules as we go :)

Thanks for thinking carefully and openly about these issues @maximede !

@suyograo @ph @jordansissel are your thoughts here?

**Comment by andrewvc** [comment]
Also @maximede to be clear, the parsing actually happens in the codec, you don't need to parse it in a filter. You will get the fully parsed and structured JSON in the 'message' hash. I wasn't sure if you were clear on how the codecs worked here.

**Comment by maximede** [comment]
Yes, I missed the part with the nested variables at first. I read the code a bit fast and I thought that we were storing the string result in the message.

**Comment by andrewvc** [comment]
I will register my preference that we keep the current format. [In-band Signalling](https://en.wikipedia.org/wiki/In-band_signaling) is usually just asking for some sort of trouble.

**Comment by ph** [comment]
jumping in in this discussion.

I think this plugin is really similar to this one https://github.com/logstash-plugins/logstash-input-http, by default the plugin will replace the event with the unserialized json content of the request.body, to be consistent this plugin should have the same behavior.

For other values like `name` we can use the `@metadata` field, by default this key wont be serialized in the output, if you look at implementation for the https://github.com/logstash-plugins/logstash-input-elasticsearch, you can see that we save the `docinfo` for the document in the `@metadata` field but we also allow the user to override this configuration.

I've looked a bit in the code and I think instead of manually specifying what we want to add to the event we can directly save the request headers in the `@metadata`, the headers will contains all the information we need for future work.

Also we can remove the "success" logic, this business logic can be replicated with conditionals and the mutate filters:

```
event["success"] = response.code >= 200 && response.code <= 299
```

What do you think?

**Comment by andrewvc** [comment]
@ph that makes perfect sense. I'll move all the metadata around the HTTP request to the `@metadata` field. Also, agreed with the `success` field.

**Comment by andrewvc** [comment]
Fixed in #10
